features/home/providers/vestings-provider.tsx

```diff
@@ -19,47 +20,73 @@ export const iconsMap = {
 };
 
 export type VestingsValue = {
-  currentVesting?: string;
-  vestings?: string[];
   setCurrentVesting: (vesting: string) => void;
-  isClaiming: boolean;
-  setIsClaiming: (value: boolean) => void;
-  isVestingsLoading: boolean;
-};
+} & (
+  | {
+      isVestingsLoading: false;
+      currentVesting: string;
+      vestings: string[];
+    }
+  | {
+      isVestingsLoading: false;
+      currentVesting: undefined;
+      vestings: [];
+    }
+  | {
+      isVestingsLoading: true;
+      currentVesting: undefined;
+      vestings: undefined;
+    }
+);
 
-const VestingsProvider: FC<PropsWithChildren> = ({ children }) => {
+export const VestingsProvider: FC<PropsWithChildren> = ({ children }) => {
   const [currentVesting, setCurrentVesting] = useState<string>();
   const [vestings, setVestings] = useState<string[]>();
-  const [isClaiming, setIsClaiming] = useState(false);
+  const [isVestingsLoading, setIsVestingsLoading] = useState<boolean>(true);
```

Comment:
Why do you need one more loading state `isVestingsLoading`? Is `isLoading` from `useVestings` not enough?

Reply:
May be not, will check

Reply:
Good catch, it was not actually needed, removed